cli/cmd/add_resource.go

```diff
@@ -86,20 +88,66 @@ keptn add-resource --project=rockshop --stage=production --service=shop --resour
+
+		// Prevent setting --stage and --all-stages at the same time
+		if (addResourceCmdParams.Stage != nil  && *addResourceCmdParams.Stage != "") &&
+			(addResourceCmdParams.AllStages != nil && *addResourceCmdParams.AllStages) {
+			return errors.New("Cannot use --stage and --all-stages at the same time")
+		}
+
+		// When setting --all-stages, project and service needs to be set
+		if (addResourceCmdParams.AllStages != nil && *addResourceCmdParams.AllStages) && (
+			(addResourceCmdParams.Service == nil || *addResourceCmdParams.Service == "") || (addResourceCmdParams.Project == nil || *addResourceCmdParams.Project == "")) {
+			return errors.New("--service and --project need to be supplied when using --all-stages")
+		}
+
+
+
+		// Handle different cases of adding resource to a projects default branch, stage branch, and/or service sub-directory
+		if (addResourceCmdParams.Service != nil && *addResourceCmdParams.Service != "") && (addResourceCmdParams.AllStages != nil && *addResourceCmdParams.AllStages) {
+			// add to all stages
+			logging.PrintLog("Adding resource "+*addResourceCmdParams.Resource+" to all stages of project "+*addResourceCmdParams.Project, logging.InfoLevel)
+		} else if (addResourceCmdParams.Service != nil && *addResourceCmdParams.Service != "") && (addResourceCmdParams.Stage != nil && *addResourceCmdParams.Stage != "") {
+			// add to service and stage
 			logging.PrintLog("Adding resource "+*addResourceCmdParams.Resource+" to service "+*addResourceCmdParams.Service+" in stage "+*addResourceCmdParams.Stage+" in project "+*addResourceCmdParams.Project, logging.InfoLevel)
 		} else if (addResourceCmdParams.Service == nil || *addResourceCmdParams.Service == "") && (addResourceCmdParams.Stage != nil && *addResourceCmdParams.Stage != "") {
+			// service is empty, add to stage
 			logging.PrintLog("Adding resource "+*addResourceCmdParams.Resource+" to stage "+*addResourceCmdParams.Stage+" in project "+*addResourceCmdParams.Project, logging.InfoLevel)
 		} else if (addResourceCmdParams.Service == nil || *addResourceCmdParams.Service == "") && (addResourceCmdParams.Stage == nil || *addResourceCmdParams.Stage == "") {
+			// service and stage are empty, add to default branch
 			logging.PrintLog("Adding resource "+*addResourceCmdParams.Resource+" to project "+*addResourceCmdParams.Project, logging.InfoLevel)
 		} else {
 			return errors.New("Flag 'stage' is missing")
 		}
 
 		if !mocking {
-			_, errorObj := resourceHandler.CreateResources(*addResourceCmdParams.Project, *addResourceCmdParams.Stage, *addResourceCmdParams.Service, resources)
-			if errorObj != nil {
-				return errors.New("Resource " + *addResourceCmdParams.Resource + " could not be uploaded: " + *errorObj.Message)
+			if addResourceCmdParams.AllStages != nil && *addResourceCmdParams.AllStages {
+				// Upload to all stages
+
+				// get stages
+				stagesHandler := apiutils.NewAuthenticatedStageHandler(endPoint.String(), apiToken, "x-token", nil, endPoint.Scheme)
```

Comment:
please append `/shipyard-controller` to the endpoint, i.e.:

```golang
stagesHandler := apiutils.NewAuthenticatedStageHandler(endPoint.String()+"/shipyard-controller", apiToken, "x-token", nil, endPoint.Scheme)
```

Reply:
I've changed it, but I believe we should make sure this is reflected in go-utils rather than in the code that uses it.

Reply:
thanks, but I agree with your point - this will be done in https://github.com/keptn/keptn/issues/3001